src/extension/background-script/actions/cache/getCurrencyRate.ts

```diff
@@ -0,0 +1,85 @@
+import axios from "axios";
+import dayjs from "dayjs";
+import isSameOrBefore from "dayjs/plugin/isSameOrBefore";
+import browser from "webextension-polyfill";
+import { CURRENCIES } from "~/common/constants";
+import { get as getSettings } from "~/extension/background-script/actions/settings";
+import type { MessageCurrencyRateGet } from "~/types";
+
+dayjs.extend(isSameOrBefore);
+
+const storeCurrencyRate = async (rate: number) => {
+  const currencyRate = {
+    rate,
+    timestamp: Date.now(),
+  };
+
+  await browser.storage.local.set({
+    currencyRate: JSON.stringify(currencyRate),
+  });
+};
+
+const getFiatBtcRate = async (currency: CURRENCIES): Promise<number> => {
+  const result = await getSettings();
```

Comment:
as you're here in the background script you can use the state directly and don't need to go through the messaging

Reply:
Thanks! Done